src/RA_Dlg_RomChecksum.cpp

```diff
@@ -2,53 +2,76 @@
 
 #include "RA_Resource.h"
 #include "RA_Core.h"
+#include "ra_wm_handler.h"
 
-//static 
-BOOL RA_Dlg_RomChecksum::DoModalDialog()
+namespace ra {
+namespace ui {
+
+INT_PTR Dlg_RomChecksum::DoModal()
 {
-    return DialogBox(g_hThisDLLInst, MAKEINTRESOURCE(IDD_RA_ROMCHECKSUM), g_RAMainWnd, RA_Dlg_RomChecksum::RA_Dlg_RomChecksumProc);
+    // ::GetAncestor(::GetActiveWindow(), GA_ROOT) is needed since
+    // GetActiveWindow might return the menu item instead of g_RAMainWnd, it
+    // means that it get's the root window.
+    return DialogBoxParam(::GetModuleHandle(_T("RA_Integration.dll")), MAKEINTRESOURCE(IDD_RA_ROMCHECKSUM),
+                          ::GetAncestor(::GetActiveWindow(), GA_ROOT), Dlg_RomChecksumProc,
+                          reinterpret_cast<LPARAM>(this));
 }
 
-INT_PTR CALLBACK RA_Dlg_RomChecksum::RA_Dlg_RomChecksumProc(HWND hDlg, UINT nMsg, WPARAM wParam, LPARAM lParam)
+struct ChecksumHelper : public WM_Handler
 {
-    switch (nMsg)
+    BOOL OnInitDialog(HWND hwnd, _UNUSED HWND hwndFocus, _UNUSED LPARAM lParam) noexcept override
     {
-        case WM_INITDIALOG:
-            SetDlgItemText(hDlg, IDC_RA_ROMCHECKSUMTEXT, NativeStr(std::string("ROM Checksum: ") + g_sCurrentROMMD5).c_str());
-            return FALSE;
+        // It seems hwndFocus is the thing with the default focus, (usually OK button)
+        using WindowH = std::unique_ptr<HWND__, decltype(&::DestroyWindow)>;
+        WindowH hRomChecksum{ ::GetDlgItem(hwnd, IDC_RA_ROMCHECKSUMTEXT), ::DestroyWindow };
+
+        // just pray it doesn't leak, we can't destroy it or we won't see anything
+        OnSetText(hRomChecksum.release(), NativeStr("ROM Checksum: " + g_sCurrentROMMD5).c_str());
```

Comment:
We'd need to destroy during `WM_NCDESTROY `but it wasn't used before so I didn't but it in. This is just a POC.